Preprocessing/oxygenPP.m

```diff
@@ -0,0 +1,200 @@
+% INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN 
+% CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) 
+% ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE 
+% POSSIBILITY OF SUCH DAMAGE.
+%
+narginchk(2, 3);
+
+if ~iscell(sample_data), error('sample_data must be a cell array'); end
+if isempty(sample_data), return;                                    end
+
+% no modification of data is performed on the raw FV00 dataset except
+% local time to UTC conversion
+if strcmpi(qcLevel, 'raw'), return; end
+
+% auto logical in input to enable running under batch processing
+if nargin<3, auto=false; end
+
+for k = 1:length(sample_data)
+  
+  sam = sample_data{k};
+  
+  % Get the temperature
+  tempIdx       = getVar(sam.variables, 'TEMP');
+  if ~(tempIdx), continue; end % if there is no temperature, then don't do anythin  
+  % get coorinates and dimensions from TEMP to use as output
+  % coordinates/dimensions
+  dimensions = sam.variables{tempIdx}.dimensions;
+  if isfield(sam.variables{tempIdx}, 'coordinates')
+      coordinates = sam.variables{tempIdx}.coordinates;
+  else
+      coordinates = '';
+  end
+  temp = sam.variables{tempIdx}.data;
+  
+  % indexs of other variables that might be needed
+  psalIdx       = getVar(sam.variables, 'PSAL');
+  
+  [presRel, presName] = presRelFromSampleData(sam);
+    
+  % oxygen parameters that maybe in the dataset
+  doxIdx        = getVar(sam.variables, 'DOX');
+  dox1Idx       = getVar(sam.variables, 'DOX1');
+  dox2Idx       = getVar(sam.variables, 'DOX2');
+  doxsIdx       = getVar(sam.variables, 'DOXS');
+  oxsolIdx      = getVar(sam.variables, 'OXSOL');
+  
+  % check is psal and some dox measurement present, if not skip
+  if ~(psalIdx && (doxIdx || dox1Idx || dox2Idx || doxsIdx)), continue; end
+    
+  psal = sam.variables{psalIdx}.data;
+  
+  ptmp = sw_ptmp(psal, temp , presRel, 0);
+  
+  % calculate the density at 0 dbar
+  dens = sw_dens0(psal, ptmp);
+
+  % calculate oxygen solability
+  if ~(oxsolIdx)
+      oxsol = gsw_O2sol_SP_pt(psal, ptmp); % Should this be potential temperature?
+      oxsolComment = 'oxsolPP.m: derived from PSAL, TEMP using Garcia and Gordon (1992, 1993), calculated using teos-10 function gsw_O2sol_SP_pt(SP,pt)';
+  else
+      oxsol = sam.variables{oxsolIdx}.data;
+  end
+    
+  if (doxIdx) && ~(dox2Idx)
+    dox = sam.variables{doxIdx}.data;
+    dox2 = dox * 44659.6 ./ dens;
+    dox2Comment = 'oxsolPP.m: derived from dox2 = dox (ml/l) * 44659.6 / sw_dens0(PSAL, PTEMP)';
+  end
+  if (dox1Idx) && ~(dox2Idx)
+    dox1 = sam.variables{dox1Idx}.data;
+    dox2 = dox1 ./ dens;
+    dox2Comment = 'oxsolPP.m: derived from dox2 = dox (umol/l) / sw_dens0(PSAL, PTEMP)';
+  end
+  if (doxsIdx) && ~(dox2Idx)
+    doxs = sam.variables{doxsIdx}.data;
+    dox2 = doxs .* oxsol;
+    dox2Comment = 'oxsolPP.m: derived from doxs * oxsol(PSAL, TEMP)';
+  end
```

Comment:
do we have reason be believe the manufacture calculates one better than the other? I would chose DOX (ml/l) from a SBE43 because that is what they are calibrated in. This would apply to a WQM data also. Optode they are calibrated in umol/l so DOX1 would be the preference.